### PR TITLE
year of the cow redo

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/bronze/usaco-1107.mdx [hunk 1]
```diff
@@ -5,144 +5,196 @@ title: Year of the Cow
...
-		ans = arr[i][7]
-		break
-
-print(abs(ans))
+	std::map<string, int> birthYears{{"Bessie", 0}};
+	for (Relation r : relations) {
+		int change = r.prev ? -1 : 1;
+		// +change because it has to be at least 1 year off
+		int year = birthYears[r.relative] + change;
+		while (mod(year, ZODIAC.size()) != r.year) {
```
Comment (envyaims) [inline solutions/bronze/usaco-1107.mdx:71]: why a function

**Review by envyaims — APPROVED** [review]
poop

**Review by ryanchou-dev — APPROVED** [review]
okkk

File: solutions/bronze/usaco-1107.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/bronze/usaco-1107.mdx:71]: it's too long if i put it inside

**Pull request merged**